src/sliceDataStorage.cpp

```diff
@@ -469,4 +477,113 @@ bool SliceDataStorage::getExtruderPrimeBlobEnabled(int extruder_nr) const
     return train->getSettingBoolean("prime_blob_enable");
 }
 
+
+void SupportLayer::excludeAreasFromSupportInfillAreas(const Polygons& exclude_polygons, const AABB& exclude_polygons_boundary_box)
+{
+    // record the indexes that need to be removed and do that after
+    std::vector<size_t> to_remove_indexes;  // FIFO for replacing, LIFO for removing
+    to_remove_indexes.reserve(this->support_infill_parts.size());
+    size_t to_remove_indexes_front_idx = 0;
+
```

Comment:
This algorithm has become so much larger than I intended.
Maybe it's a bit overengineered now because we talked so much about it.
😅 

Let me think about a different way of doing it.